lib/_http_client.js

```diff
@@ -68,6 +68,11 @@ function ClientRequest(options, cb) {
   self.socketPath = options.socketPath;
   self.timeout = options.timeout;
 
+  const _method = options.method;
+  if (_method !== undefined && _method !== null
+    && typeof _method !== 'string') {
+    throw new TypeError('Method must be a string');
+  }
```

Comment:
For the best backwards compatibility, I think the `undefined`/`null` check should be removed and just test for truthiness:


```js
if (options.method && typeof options.method !== 'string') {
  throw new TypeError('Method must be a string');
}
```